compiler/rustc_hir/src/hir.rs

```diff
@@ -1848,10 +1848,10 @@ pub enum ExprKind<'hir> {
     /// This construct only exists to tweak the drop order in HIR lowering.
     /// An example of that is the desugaring of `for` loops.
     DropTemps(&'hir Expr<'hir>),
-    /// A `let $pat = $expr` expression.
-    ///
-    /// These are not `Local` and only occur as expressions.
+    /// These are not [`Local`] and only occur as expressions.
     /// The `let Some(x) = foo()` in `if let Some(x) = foo()` is an example of `Let(..)`.
+    ///
+    /// A `let $pat = $expr` expression.
```

Comment:
we have this at the start of docs for all other enum variants, let's keep it there. The second line in the docs clears things up after all

Reply:
Yes, and it's the issue here. When looking for `let` (the statement), we have (now) the choice between `Expr` and `StmtKind`. Clearing definitely this confusion with the documentation would be nice imo. Feel free to close the PR if you disagree though, I think the `StmtKind` variant renaming already improved the situation.